src/libs/DistanceRequestUtils.ts

```diff
@@ -45,7 +46,7 @@ function getMileageRates(policy: OnyxInputOrEntry<Policy>, includeDisabledRates
     }
 
     Object.entries(distanceUnit.rates).forEach(([rateID, rate]) => {
-        if (!includeDisabledRates && rate.enabled === false) {
+        if (!includeDisabledRates && rate.enabled === false && (!selectedRateID || rateID !== selectedRateID)) {
```

Comment:
`!selectedRateID` would fail if we have a rateID that's `"0"` no?

Reply:
No, `"0"` would be the ID, which means it has an ID

Reply:
If I create a rate whose ID is the string `"0"` then select it, this would not work because `!"0"` is false but that's wrong because we indeed passed an ID, so we should've compared `selectedRateID` with `rateID` but we did not

Reply:
Oh, you are right! I was confusing the behavior with PHP where `"0"` is falsy